File: 0024_Swap Nodes in Pairs/SwapNodesinPaires_Recursion.java
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head == null || head.next == null) return head;
        ListNode p = head.next;
        head.next = swapPairs(head.next.next);
        p.next = head;
        return p;
    }
}

File: 0024_Swap Nodes in Pairs/SwapNodesinPaires_Recursion.cs
public class Solution {
    public ListNode SwapPairs(ListNode head) {
        if(head == null || head.next == null) return head;
        var p = head.next;
        head.next = SwapPairs(head.next.next);
        p.next = head;
        return p;
    }
}